docs/tutorials/Containers.md

```diff
@@ -0,0 +1,475 @@
+        inputBinding:
+          prefix: "Hello, "
+outputs:
+    out: stdout
+
+requirements:
+    DockerRequirement:
+        dockerPull: "ubuntu:latest"
+``` 
+___
+### Docker
+
+[Docker](https://www.docker.com) is a popular container technology that is natively supported by Cromwell and WDL.
+
+
+#### Docker on a Local Backend
+
+On a single machine (laptop or server), no extra configuration is needed to allow docker to run, provided Docker is installed.
+
+You can install Docker for Linux, Mac or Windows from [Docker Hub](https://hub.docker.com/search/?type=edition&offering=community)
+
+#### Docker on Cloud
+
+It is strongly advised that you provide a Docker image to tasks that will run on Cloud backends, and in fact most Cloud providers require it.
+
+It might be possible to use an alternative container engine, but this is not recommended if Docker is supported.
+
+#### Docker on HPC
+
+Docker can allow running users to gain superuser privileges, called the [Docker daemon attack surface](https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface). In HPC and multi-user environments, Docker recommends that "only trusted users should be allowed to control your Docker Daemon".
+
+For this reason it's worth exploring other technologies that will support the reproducibility and simplicity of running a workflow that uses docker containers; Singularity or udocker.
+
+___
+
+### Singularity
+
+Singularity is a container engine designed for use on HPC systems in particular, while ensuring an appropriate level of security that Docker cannot provide.
+
+#### Installation
+Before you can configure Cromwell on your HPC system, you (or your sysadmin) will have to install Singularity, which is documented [here](https://www.sylabs.io/guides/3.0/admin-guide/admin_quickstart.html#installation).
+Installation of Singularity doesn't require root, but to access the full features of Singularity, such as the use of its preferred image format, the Singularity binary must be owned by root, with the `setuid` bit enabled.
+This is [documented here](https://www.sylabs.io/guides/2.6/admin-guide/security.html#how-does-singularity-do-it).
+If you do not give Singularity these privileges, you are restricted to using sandbox images, which come with a number of downsides.
+Because this is not an ideal way to run Singularity, you might consider forwarding [this letter](https://www.sylabs.io/guides/3.0/user-guide/installation.html#singularity-on-a-shared-resource) to your admins.
+
+#### Configuring Cromwell for Singularity
+
+Once Singularity is installed, the main configuration block you'll need to modify is the `config` block inside `backend.providers` in your Cromwell configuration.
+In particular, this block contains a key called `submit-docker`, which will contain a script that is run whenever a job needs to run that uses a Docker image.
+If the job does not specify a Docker image, then the regular `submit` block will be used instead.
+
```

Comment:
My expectation is to see the example config here.

Reply:
I did a bit of restructuring in this section to take advantage of some of the ideas you raised through your comments below to. I moved the caching stuff up to the top, reworded a few things to better lead into the configs and added some stronger motivations for the choices in the config. I'll reply to your comments below, but they may appear out of order on this PR.